hisyonosuke/src/shift-changer/shift-changer.ts

```diff
@@ -0,0 +1,438 @@
+  const dataColumn = sheet.getLastColumn();
+  sheet.getRange(6, 1, dataRow, dataColumn).clearContent();
+
+  const payload = {
+    external_id: "shift-changer",
+    operationType: "showEvents",
+    userEmail: userEmail,
+    startDate: startDate,
+  };
+  const options: GoogleAppsScript.URL_Fetch.URLFetchRequestOptions = {
+    method: "post",
+    payload: payload,
+  };
+  const {API_URL} = getConfig();
+  const response = UrlFetchApp.fetch(API_URL, options);
+  if (!response.getContentText()) return;
+
+  const eventInfos: { title: string; date: string; startTime: string; endTime: string }[] = JSON.parse(
+    response.getContentText()
+  );
+  const moldedEventInfos = eventInfos.map((eventInfo) => {
+    return [eventInfo.title, eventInfo.date, eventInfo.startTime, eventInfo.endTime];
+  });
+  
+  sheet.getRange(6, 1, moldedEventInfos.length, moldedEventInfos[0].length).setValues(moldedEventInfos);
+};
+
+const getSheet = (operationType: OperationType, spreadsheetUrl: string): GoogleAppsScript.Spreadsheet.Sheet => {
+  const today = format(new Date(), "yyyy-MM-dd");
+  const sheet = SpreadsheetApp.openByUrl(spreadsheetUrl)
+    .getSheets()
+    .find((sheet) => sheet.getDeveloperMetadata()[0].getKey() === `${today}-${operationType}`);
+
+  if (!sheet) throw new Error("SHEET is not defined");
+
+  return sheet;
+};
+
+const getShiftInfos = (spreadsheetUrl: string) => {
+  const operationType = "registration";
+  const sheet = getSheet(operationType, spreadsheetUrl);
+  const lastRowNum = sheet.getLastRow();
+  const shiftInfos = sheet.getRange(2, 1, lastRowNum - 1, 6).getValues();
+  return shiftInfos;
+};
+
+const createTitleFromShiftInfo = (
+  shiftInfo: any[],
+  userEmail: string,
+  slackMemberProfiles: {
+    name: string;
+    email: string;
+  }[]
+): string => {
+  const name = getNameFromEmail(userEmail, slackMemberProfiles);
+  const nameRegex = new RegExp(name.replace(/ |\u3000/g, "( |\u3000|)?"));
+  const job = getJob(nameRegex);
+
+  const workingStyle = shiftInfo[5];
+
+  if (shiftInfo[3] === "" || shiftInfo[4] === "") {
+    const title = `【${workingStyle}】${job}: ${name}さん`;
+    return title;
+  } else {
+    const restStartTime = format(shiftInfo[3], "HH:mm");
+
+    const restEndTime = format(shiftInfo[4], "HH:mm");
+
+    const title = `【${workingStyle}】${job}: ${name}さん (休憩: ${restStartTime}~${restEndTime})`;
+    return title;
+  }
+};
+
+const getNameFromEmail = (email: string, slackMemberProfiles: { name: string; email: string }[]): string => {
+  const slackMember = slackMemberProfiles.find((slackMemberProfile) => slackMemberProfile.email === email);
+  if (!slackMember) throw new Error("The email is non-slack member");
+  return slackMember.name;
+};
+
+const getSlackMemberProfiles = (client: SlackClient): { name: string; email: string }[] => {
+  const response = client.users.list();
+  const slackMembers = response.members;
+  if (!slackMembers) throw new Error("SLACK_MEMBERS is not defined");
```

Comment:
後続でmapやfilter等の配列操作をする場合、undefinedは空配列にしてあげると処理が簡潔になります。
ちゃんとエラーを吐くべき場合もあるので場合によりけりですが、ここでエラーを吐く必要はないはず。

```suggestion
  const slackMembers =  client.users.list().members ?? [];
```

Reply:
対応しました b0432e54464599a562a69a783267881939c80f3b